stubs/openpyxl/openpyxl/reader/excel.pyi

```diff
@@ -44,7 +45,7 @@ class ExcelReader:
     def read(self) -> None: ...
 
 def load_workbook(
-    filename: SupportsRead[bytes] | StrPath,
+    filename: StrPath | IO[bytes],
```

Comment:
https://github.com/python/typeshed/pull/9511/files#r1113278179 Avasam 
> This, and `ExcelReader.__init__` is ultimately passed to `ZipFile` in `_validate_archive`.
`ZipFile.__init__` is typed as `ZipFile(file: StrPath | IO[bytes], ...)`
> 
> Even in read mode it still needs a `name` variable, `seek` and `tell` methods.

AlexWaygood
> I'd be okay with switching both of the conflicting annotations to `IO[bytes]`. We could probably come up with a more precise protocol, but for now consistency with the stdlib `zipfile` stub is probably the most important thing

Reply:
This change is responsible for the `mypy_primer` diff.
Affected code in Pandas: https://github.com/pandas-dev/pandas/blob/705d4312cf1d94ef2497bcd8091e0eabd1085f4a/pandas/io/excel/_openpyxl.py#L566

Reply:
Given the primer hit I think we should bite the bullet, introduce a protocol in zipfile, and use an equivalent protocol here.

Reply:
I don't mind holding off on this PR until we have a proper Protocol defined for zipfile.

I'll link #5835 as this is another use-case for it. Otherwise we'll have to duplicate the protocol in openpyxl for a while.

Reply:
I don't think #5835 is happening anytime soon so for now we'll just have to duplicate it.